### ISSUE TITLE
Assertion scan: classify unexercised nodes by cause (importer vs pass-raised) + per-node fixture guarantee

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#2237]
## Context

Follow-up to #2224 / #2225 (assertion-scan audit corpus + coverage report), surfaced while auditing annotation batch 7 (#2232) against the audit corpus.

Over 115,669 methods (22 assemblies, 0 pass bugs), the batch-7 nodes `Lambda` and `LocalFunctionInvocation` were reported **not exercised** — sitting in the same "not exercised" list as rare opcodes like `CallIndirect`/`Unbox`. But they are a *different, more actionable* signal, and the report can't distinguish them.

## Problem

The coverage report's "not exercised" list flattens three distinct causes:

1. **Rare opcode** (`CallIndirect`, `Unbox`) — genuinely uncommon in the corpus. A breadth gap; low urgency.
2. **Pass-raised recognition node** (`Lambda`, `LocalFunctionInvocation`, and later the pattern/async families) — produced only when a raising pass recognizes a source shape. If it never appears, that is either a corpus-shape gap **or a silently-not-firing raise pass** — a potential correctness regression hiding behind a coverage gap.
3. **Structurally hard to reach** — needs a construct no real-world corpus reliably contains.

A not-firing raise pass and "the corpus has no lambdas" look identical in the current report, so the highest-value coverage signal (a raise pass that stopped producing its node) is invisible.

This is also where the assertion machinery starts finding *decompiler* correctness issues rather than just annotation-description issues: a pass-raised node that a fixture *should* produce but doesn't is a real raise-pass regression.

### 1. Classify unexercised nodes by cause

Tag each annotated node as **importer-emitted** vs **pass-raised** (a node whose only producers are raising passes, not the importer), and split the "not exercised" report accordingly. An unexercised *pass-raised* node should read as "likely raise-pass or fixture gap — investigate," distinct from an unexercised rare opcode.

### 2. Per-node fixture guarantee

Give each `[InverseOf]` node a minimal compiled fixture that deterministically produces it (marrying the #2225 audit corpus with the existing `--generated-fixtures` ladder). Run these in the daily assertion scan so:

- "unexercised" can reach a true **0** (every node provably exercised by at least its fixture), and
- a fixture that **stops** producing its node becomes a **raise-pass-regression alarm** — a real correctness signal, report-only, not a gate.

## Guardrail

Same as #2224: this stays **measurement / triage**, not a correctness gate. The fixture-regression alarm is a report signal for investigation, not a CI pass/fail on a count.

## Acceptance

- Coverage report distinguishes importer-emitted vs pass-raised unexercised nodes.
- Each `[InverseOf]` node has a fixture that produces it; the daily scan reports any node its fixture no longer produces.
- Documented as measurement in `tools/DecompilerHarness/README.md`, pointing at `docs/decompiler-raise-discipline.md`.

## References

- #2224 (audit corpus + coverage follow-ups), #2225 (audit corpus report), #2232 (batch 7, where `Lambda`/`LocalFunctionInvocation` surfaced this).
- `docs/decompiler-raise-discipline.md` — the measurement-not-a-gate framing.

**Comment by richlander** [comment]
Claiming a focused first slice on branch `fix/issue-2237-assertion-scan`: classify assertion-scan unexercised nodes by producer category (importer-emitted vs pass-raised) and document it as measurement/reporting. I will leave the full per-node fixture guarantee as follow-up unless it falls out naturally.

**Comment by richlander** [comment]
Closing as already implemented on current `origin/main`.

Evidence checked:
- Assertion scan report splits unexercised nodes by cause:
  - `importer-emitted not exercised`
  - `pass-raised not exercised (raise-pass or fixture gap; investigate)`
- `InverseLedger.CauseFor(...)` classifies annotated nodes as `ImporterEmitted` vs `PassRaised`; `AssertionScanTests` includes both spot checks and an exhaustive annotated-node classification test.
- `--assertion-fixture-guarantee` exists and reports:
  - generated fixture guarantee summary,
  - nodes without fixture mappings,
  - fixture regression alarms as report-only.
- `AssertionScanTests` verifies fixture mappings exist for every annotated node and that fixture observations are counted only from declared fixtures.
- `tools/DecompilerHarness/README.md` documents the measurement/not-a-gate framing, the split unexercised lists, fixture regression alarms, and daily assertion scan usage.

Focused validation passed:

```bash
dotnet run --no-restore --project src/ILInspector.Decompiler.Tests -c Release -- -filter "/*/*/AssertionScanTests/*"
# 21/21 passed
```
